quic/s2n-quic-core/src/packet/number/tests.rs

```diff
@@ -140,25 +145,22 @@ fn rfc_decoder(largest_pn: u64, truncated_pn: u64, pn_nbits: usize) -> u64 {
     let pn_mask = pn_win - 1;
 
     let candidate_pn = (expected_pn & !pn_mask) | truncated_pn;
-    if catch(|| {
-        candidate_pn <= expected_pn.checked_sub(pn_hwin).unwrap()
-            && candidate_pn < (1u64 << 62).checked_sub(pn_win).unwrap()
-    })
-    .unwrap_or_default()
-    {
+    if catch!(
+        candidate_pn <= expected_pn.checked_sub(pn_hwin)?
+            && candidate_pn < (1u64 << 62).checked_sub(pn_win)?
+    ) {
         return candidate_pn + pn_win;
     }
 
-    if catch(|| candidate_pn > expected_pn.checked_add(pn_hwin).unwrap() && candidate_pn >= pn_win)
-        .unwrap_or_default()
-    {
+    if catch!(candidate_pn > expected_pn.checked_add(pn_hwin)? && candidate_pn >= pn_win) {
```

Comment:
did this have any performance benefit or is it mostly stylistic/simplification?

Reply:
This is to support Kani execution, as it doesn't support catching panic unwinding.